cpp/tests/transform/one_hot_encode_tests.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/table/table_view.hpp>
+#include <cudf/transform.hpp>
+
+#include <limits>
+
+namespace cudf {
+namespace test {
+
+template <typename T>
+struct OneHotEncodingTestTyped : public BaseFixture {
+};
+
+struct OneHotEncodingTest : public BaseFixture {
+};
+
+TYPED_TEST_CASE(OneHotEncodingTestTyped, cudf::test::NumericTypes);
+
+TYPED_TEST(OneHotEncodingTestTyped, Basic)
+{
+  auto input    = fixed_width_column_wrapper<int32_t>{8, 8, 8, 9, 9};
+  auto category = fixed_width_column_wrapper<int32_t>{8, 9};
+
+  auto col0 = fixed_width_column_wrapper<bool>{1, 1, 1, 0, 0};
+  auto col1 = fixed_width_column_wrapper<bool>{0, 0, 0, 1, 1};
+
+  auto expected = table_view{{col0, col1}};
+
+  auto [_, got] = one_hot_encode(input, category);
```

Comment:
Rather than using an underscore I would recommend actually naming the column variable and using the `maybe_unused` attribute. When I see an underscore I think you're throwing away the value, but really you need to keep the column in scope since that's what owns the data. `maybe_unused` better expresses the fact that the variable needs to exist, it just may never be accessed. Unfortunately there's no way to apply it to just one returned value from a structured binding, but I still think that it more clearly expresses intent in this case.

Reply:
That's an awesome suggestion. Should I decorate [[maybe_unused]] before auto in this case?